### PR TITLE
docs: add description of what Ignition is

### PR DESCRIPTION
**Pull request by omkensey**
Add a short description of how Ignition operates.

### CODE REVIEW

**Comment by brianredbeard** [comment]
[:+1:](http://lgtm.in/u/I8T0fYsPp) [:-1:](http://lgtm.in/r/I8T0fYsPp)

File: doc/getting-started.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Getting Started with Ignition #
 
+## What is Ignition? ##
+
+Ignition is a boot-time configuration utility that runs before the switch out of the initramfs during boot.  Mechanically-speaking, it is an executable in the initramfs that looks for configuration data (or a pointer to a URL for such data) in an environment-specific location and applies it to the machine before switch_root is called (see below under [Providing a Config](#providing-a-config) for details).
```
Comment (joshix) [inline doc/getting-started.md]: Ignition is a low-level system configuration utility. The Ignition executable is part of the temporary initial root filesystem, the *initramfs*. When Ignition runs, it finds configuration data in a named location for a given environment, such as a file or URL, and applies it to the machine before `switch_root` is called to mount the machine's root filesystem.

**Comment by omkensey** [comment]
I like @joshix's revision.

**Comment by joshix** [comment]
@omkensey Since it is in your fork, we can do this 1 of 2 ways:
1) The usual -- just adopt my suggestions into your branch, and rebase to squash down to one commit, then I'll re-review and merge 
2) I can close-supersede and go directly to my changes. This will be later today or tomorrow.

**Comment by joshix** [comment]
Closed in favor of #195

**Pull request closed without merging**